src/lib/dlt_user.c

```diff
@@ -3827,13 +3827,18 @@ void dlt_user_housekeeperthread_function(void *ptr)
 
     pthread_cleanup_push(dlt_user_cleanup_handler, NULL);
 
+
+    pthread_mutex_lock(&dlt_housekeeper_running_mutex);
+
     // signal dlt thread to be running
     *dlt_housekeeper_running = true;
```

Comment:
Hello @ranoongs
The variable you trying to lock is an atomic type and it is totally free from race data.
Hence I believe we do not need to lock the variable.
Kindly check this point.

Reply:
Yes, as you said, dlt_housekeeper_running is atomic.

The objective for this commit is to avoid unnecessary pthread_cond_timedwait's timeout.



![1](https://github.com/user-attachments/assets/19b1e909-e84b-4df9-8049-de8de49da06c)